ibis_substrait/compiler/translate.py

```diff
@@ -937,9 +937,13 @@ def _contains(
     compiler: SubstraitCompiler,
     **kwargs: Any,
 ) -> stalg.Expression:
+    if not isinstance(op.options, (collections.abc.Sequence, ir.ListExpr)):
```

Comment:
Is this to support column expressions? If so, I'm not sure if `SingularOrList` is the right mapping. I think you'd probably want a `Subquery` since the corresponding SQL would be something like

```sql
SELECT a IN (SELECT b FROM t)
FROM t
```

That would be something like:

```python
Subquery(
    in_predicate=InPredicate(
        needles=[translate(op.value, compiler, **kwargs)],
        haystack=translate(op.options.to_projection(), compiler, **kwargs),
    )
)
```

Reply:
It's coming up in situations like this one:
```
In [16]: ~partsupp.ps_suppkey.isin(
    ...:     supplier.filter([supplier.s_comment.like("%Customer%Complaints%")]).s_suppkey
    ...: )
Out[16]: 
r0 := UnboundTable: supplier
  s_suppkey   int32
  s_name      string
  s_address   string
  s_nationkey int32
  s_phone     string
  s_acctbal   decimal(prec=15, scale=2)
  s_comment   string

r1 := UnboundTable: partsupp
  ps_partkey    int32
  ps_suppkey    int32
  ps_availqty   int32
  ps_supplycost decimal(prec=15, scale=2)
  ps_comment    string

r2 := Selection[r0]
  predicates:
    r0.s_comment LIKE '%Customer%Complaints%'

Not(Contains(value=r1.ps_suppkey, options=r2.s_suppkey))
```

so there's only a single "option" but that option is a selection

Reply:
and looking at your example again, I see that yes, we probably want a subquery. 🥴